web/src/main/java/no/nav/foreldrepenger/fordel/web/app/rest/journalføring/FerdigstillJournalføringTjeneste.java

```diff
@@ -0,0 +1,375 @@
+                throw new TekniskException("FP-15689", lagUgyldigInputMelding("Bruker", BRUKER_MANGLER));
+            }
+        }
+
+        final var forsendelseId = asUUID(journalpost.getEksternReferanseId());
+
+        String eksternReferanseId = null;
+        if (DokumentTypeId.INNTEKTSMELDING.equals(dokumentTypeId)) {
+            eksternReferanseId =
+                journalpost.getEksternReferanseId() != null ? journalpost.getEksternReferanseId() : arkivTjeneste.hentEksternReferanseId(
+                    journalpost.getOriginalJournalpost()).orElse(null);
+        }
+
+        var mottattTidspunkt = Optional.ofNullable(journalpost.getDatoOpprettet()).orElseGet(LocalDateTime::now);
+
+        final var xml = hentDokumentSettMetadata(saksnummer, behandlingTema, aktørIdFagsak, journalpost);
+        klargjører.klargjør(xml, saksnummer, journalpostId.getVerdi(), dokumentTypeId, mottattTidspunkt, behandlingTema, forsendelseId.orElse(null),
+            dokumentKategori, enhetId, eksternReferanseId);
+
+        // For å unngå klonede journalposter fra GOSYS - de kan komme via Kafka
+        dokumentRepository.lagreJournalpostLokal(journalpost.getJournalpostId(), journalpost.getKanal(), "ENDELIG", journalpost.getEksternReferanseId());
+
+        opprettFerdigstillOppgaveTask(oppgaveId);
+    }
+
+    private ArkivJournalpost hentJournalpost(String arkivId) {
+        try {
+            return arkivTjeneste.hentArkivJournalpost(arkivId);
+        } catch (Exception e) {
+            LOG.warn("FORDEL fikk feil fra hentjournalpost: ", e);
+            throw new TekniskException("FP-15676", lagUgyldigInputMelding("Journalpost", "Finner ikke journalpost med id " + arkivId));
+        }
+    }
+
+    private static Optional<UUID> asUUID(String eksternReferanseId) {
+        try {
+            return Optional.of(UUID.fromString(eksternReferanseId));
+        } catch (Exception e) {
+            return Optional.empty();
+        }
+    }
+
+    public Optional<String> utledJournalpostTittelOgOppdater(List<DokumenterMedNyTittel> dokumenterMedNyTittel, JournalpostId journalpostId) {
+
+        final var arkivJournalpost = hentJournalpost(journalpostId.getVerdi());
+
+        var opprinneligHovedType = arkivJournalpost.getHovedtype();
+        var dokumenterFraArkiv = arkivJournalpost.getOriginalJournalpost().dokumenter();
+
+        Set<DokumentTypeId> dokumenttyper = utledDokumentTyper(dokumenterMedNyTittel, dokumenterFraArkiv);
+
+        var nyHovedtype = ArkivUtil.utledHovedDokumentType(dokumenttyper);
+        var nyJournalpostTittel = skalOppdatereJournalpostTittel(nyHovedtype, opprinneligHovedType);
+        var dokumenterÅOppdatere = mapDokumenterTilOppdatering(dokumenterMedNyTittel);
+
+        arkivTjeneste.oppdaterJournalpostMedTitler(arkivJournalpost.getJournalpostId(), nyJournalpostTittel, dokumenterÅOppdatere);
+        return nyJournalpostTittel;
+    }
+
+
+    public record DokumenterMedNyTittel(String id, String tittel) {
+
```

Comment:
Kalle id for dokumentId og tittel for dokumenttittel? Omstendelig men lettlest

Reply:
oppdatert